docs/DeveloperGuide.md

```diff
@@ -1,34 +1,663 @@
+    Figure 10b - Opening the Text Editor
+</p>
+
+&nbsp;
+
+### Loading & Storing of Data
+
+This section covers how the _Storage_ component works, from the loading of all module codes to the loading of individual module and creation of data files.
+
+#### Saving of Data
+
+The `Writer` class is responsible for writing any changes to the module’s data file, as well as creating the file itself. Interaction with the `Writer` class is done through the `ModuleList` class, whose methods are called by the other components of the app.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/writeModule.png" alt="writeModule() Sequence Diagram"><br>
+    Figure 11 - writeModule() Sequence Diagram
+</p>
+
+Whenever some data in a module changes, the command that made those changes would call the method `writeModule()` in `ModuleList` to update the change in the data file. This method would then call a method of the same name in the `Writer` class, which overwrites the existing data in the file with the new data.
+
+Due to how much data needs to be written each time, we decided to split the data file by module. That way, we only need to overwrite the module's data when changes are made.
+
+#### Loading of Data
+
+The `Loader` class is responsible for identifying all the modules currently added, as well as loading the data file of the selected class. Methods in the `Loader` class are accessed by the other components via the `ModuleList` class.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/loadModuleCode.png" alt="loadModuleCodes() Sequence Diagram"><br>
+    Figure 12 - loadModuleCodes() Sequence Diagram
+</p>
+
+To identify modules in the “Data” directory, Duke would call `loadModuleCodes()` method in the `ModuleList`. This method would then call the `getModules()` method in `Loader`, which returns a list of module codes. For each of the identified module code, `ModuleList` would call its own `insertModule()` method to add it to the module list.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/setSelectedModule.png" alt="setSelectedModule() Sequence Diagram"><br>
+    Figure 13 - setSelectedModule() Sequence Diagram
+</p>
+
+When a module is selected via the `setSelectedModule()` method, the specified module code would be searched for in the module list. If it is inside, `loadModule()` method in the `Loader` would be called. This method reads the module’s data file for data and adds them into a new instance of `Module` class. This `Module` is then returned to `ModuleList` and set as the selected module.
+
+If the `Loader` failed to load the file, null would be returned. If null is not returned, `ModuleList` would sort the data and then use `Writer` to override the existing file. This is done to remove invalid entries that were initially in the file.
+
+&nbsp;
+
+### Future Features
+
+1. Add weightage for modules.
```

Comment:
Maybe you can elaborate more about the rationale/need of adding these future improvements.